fix: check if transient exists in DB [Codeinwp/hestia-pro#2400]

cristian-ungureanu opened the pull request:
It seems like in some cases the argument of `theme_update_transient` function is false. This may happen if you delete all transients from DB and have PHP >= 8.0.x or in my case I've just changed the PHP version to 8.0 and I encounter this issue.
In this PR I just make sure the `$value` is not empty and check if the property exists.

It should fix [Codeinwp/hestia-pro#2400]

abaicus commented on src/Modules/Licenser.php:
```diff
@@ -633,7 +633,7 @@ public function update_nag() {
 	public function theme_update_transient( $value ) {
 		$update_data = $this->check_for_update();
-		if ( $update_data ) {
+		if ( ! empty( $value ) && property_exists( $value, 'response' ) && $update_data ) {
```
@cristian-ungureanu Just a suggestion here: could we do early returns for these instead of chaining all the conditions?

abaicus reviewed (commented):
@cristian-ungureanu, I dug around a bit on this. The fix should be suitable for some edge cases.

The issue happens on Hestia Pro and not on Neve Pro because the loading order is different for themes vs. plugins. That being said, the `update_themes` site transient should rarely be empty, as the core constantly updates it, and it doesn't get removed on usual transient removal (for example, `wp transient delete --all` doesn't remove site transients).

Did you manually remove it from the DB? What are the exact steps to replicate this as a user?

cristian-ungureanu commented:
@abaicus so what I did here was to update PHP from 7.4 to 8.0 on my local site.
There were lots of other notices as WP is not yet compatible with PHP 8. I've switched WP_DEBUG to false and logged in. Then the fatal error appeared and I saw the same error that Poonam reported [here](https://github.com/Codeinwp/hestia-pro/issues/2400#issue-1005182847).

I've checked the `$value` variable and it was false. I've commented the whole body of that function, refreshed the page and everything was ok. I've uncommented the code and everything was fine as now the `$value` variable was not false anymore.

In an attempt to have the same error again, I've used "Delete Expired Transients" plugin and I removed all the transients. That's how I managed to replicate it the second time.
